### PR TITLE
Merging our trajectory_generation as moveit planner

### PR DESCRIPTION
**Pull request by ct2034**
## Prerequisites
- [x] https://github.com/ros-planning/moveit/pull/2507 is released
- [x] Final run of demo_program.py
- [x] Test application template program from here: https://github.com/PilzDE/pilz_application_templates/pull/15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,13 +18,7 @@ The FTP is implemented by Pilz GmbH & Co. KG.
...
 
 ## Package: pilz_trajectory_generation
-Provides the generators to create LIN, PTP and CIRC trajectories. All of these commands can be combined and
-executed in sequence, with and without blending, using a dedicated MoveIt! capability.
-
-Custom commands can easily be defined and integrated due to the usage of a plugin structure.
-
-## Package: pilz_extensions
-Minor extensions of existing packages needed for the trajectory_generation.
+This package used to be found here but is now moved to [moveit](https://moveit.ros.org/documentation/planners/
```
Comment (jschleicher) [inline README.md]:
```suggestion
This package used to be found here but is now moved to [moveit](https://moveit.ros.org/documentation/planners/)
```

File: pilz_robot_programming/package.xml [hunk 2]
```diff
@@ -16,23 +16,22 @@
   <run_depend>moveit_commander</run_depend>
   <run_depend>tf</run_depend>
   <run_depend>rospy</run_depend>
-  <run_depend>pilz_trajectory_generation</run_depend>
+  <run_depend>pilz_industrial_motion_planner</run_depend>
   <run_depend>pilz_msgs</run_depend>
-  <run_depend>python-psutil</run_depend>
+  <run_depend>python3-psutil</run_depend>
```
Comment (jschleicher) [inline pilz_robot_programming/package.xml]: Maybe separate the python3 changes from the moveit-merge, to allow an easier backport to melodic?

File: README.md [hunk 3]
```diff
@@ -18,13 +18,7 @@ The FTP is implemented by Pilz GmbH & Co. KG.
 Relevant changes are cherry-picked into `kinetic-devel` on a case-by-case basis.
```
Comment (jschleicher) [inline README.md]: Change development branch to `noetic-devel`?

File: README.md [hunk 4]
```diff
@@ -18,13 +18,7 @@ The FTP is implemented by Pilz GmbH & Co. KG.
...
-executed in sequence, with and without blending, using a dedicated MoveIt! capability.
-
-Custom commands can easily be defined and integrated due to the usage of a plugin structure.
-
-## Package: pilz_extensions
-Minor extensions of existing packages needed for the trajectory_generation.
+This package used to be found here but is now moved to [moveit](https://moveit.ros.org/documentation/planners/
 
 ## Package: pilz\_robot\_programming
 pilz\_robot\_programming provides a python API for an intuitive programming of a MoveIt! enabled robot. For details please refer to the [ Pilz Robot Programming Documentation ](pilz_robot_programming/Readme.rst).
```
Comment (jschleicher) [inline README.md:24]: test_utils paragraph below can be dropped as well?

File: README.md (hunk 1 above)
Comment (jschleicher) [reply README.md]: A similar note should be added to [ROS wiki](http://wiki.ros.org/pilz_trajectory_generation) - probably during the follow-up PBI ROS-849?

**Review by jschleicher — APPROVED** [review]
Thank you for this work! See some inline comments.

**Comment by martiniil** [comment]
Let's revert everything we don't need for melodic and continue the noetic support over here: #331

File: pilz_robot_programming/test/integrationtests/tst_api_cmd_conversion.py [hunk 5]
```diff
@@ -417,7 +417,7 @@ def test_ptp_global_scale(self):
 
             Test results:
                 1. -
-                2. The scaled velocity scaling factor and the scaled accerleration scaling factor is set.
+                2. The scaled velocity scaling factor and the scaled accelerationscaling factor is set.
```
Comment (cdoehn) [inline pilz_robot_programming/test/integrationtests/tst_api_cmd_conversion.py]:
```suggestion
                2. The scaled velocity scaling factor and the scaled acceleration scaling factor is set.
```

Comment (martiniil) [reply pilz_robot_programming/test/integrationtests/tst_api_cmd_conversion.py]: Thanks

File: README.md (hunk 4 above)
Comment (martiniil) [reply README.md:24]: Was already dropped.

File: pilz_robot_programming/package.xml (hunk 2 above)
Comment (martiniil) [reply pilz_robot_programming/package.xml]: Yes ...

**Pull request merged**